app/models/user_annotation.rb

```diff
@@ -2,24 +2,47 @@ class UserAnnotation
   include Mongoid::Document
   field :name, type: String
   field :values, type: Array
+  field :queued_for_deletion, type: Boolean, default: false
```

Comment:
You need to implement the deletion queue functionality (use StudyFile as the example - there are attributes that have to be renamed, some can be cleared out, and some dependent models can be deleted right away)